### PR TITLE
fix docs rendering wrongly in changelog

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by JerryChen97** [comment]
@astralcai @andrijapau

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -15,22 +15,19 @@
   True
   ```
 
+<h4>New API for Qubit Mixed</h4>
+
+* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support. [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
```
Comment (JerryChen97) [inline doc/releases/changelog-dev.md:20]:
```suggestion
* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support. [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) 

```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -15,22 +15,19 @@
   True
   ```
 
+<h4>New API for Qubit Mixed</h4>
+
+* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support. [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) 
+  This module introduces an `apply_operation` helper function that features:
+
```
Comment (andrijapau) [inline doc/releases/changelog-dev.md]:
```suggestion
* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379). This module introduces an `apply_operation` helper function that features:

```

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -17,9 +17,7 @@
 
 <h4>New API for Qubit Mixed</h4>
 
-* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support. [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) 
-
-  This module introduces an `apply_operation` helper function that features:
+* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support[(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379). This module introduces an `apply_operation` helper function that features:
```
Comment (andrijapau) [inline doc/releases/changelog-dev.md:20]:
```suggestion
* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379). This module introduces an `apply_operation` helper function that features:
```

Forgot a space 😄

**Review by andrijapau — APPROVED** [review]
LGTM! 🚀

**Review by astralcai — APPROVED** [review]

**Pull request merged**